recipe_scrapers/marleyspoon.py

```diff
@@ -0,0 +1,102 @@
+import json
+import re
+import requests
+
+from ._abstract import AbstractScraper, HEADERS
+from ._exceptions import ElementNotFoundInHtml
+from ._utils import normalize_string
+
+SCRIPT_PATTERN = re.compile("gon\\.current_brand=\"(?P<brand>[^\"]+?)\".*?"
+                            "gon\\.current_country=\"(?P<country>[^\"]+?)\".*?"
+                            "gon\\.api_token=\"(?P<token>[^\"]+?)\".*?"
+                            "gon\\.api_host=\"(?P<host>[^\"]+?)\".*?")
+
+PREPARATION_DICT = {  # these values I was able to retrieve from website
+    "time_level_1": 10,  # on the website they are displayed like `5-10 minutes`, I used avg or similar rounded value
+    "time_level_2": 15,
+    "time_level_3": 20,
+    "time_level_4": 25,
+    "time_level_5": 35,
+}
+
+
+class MarleySpoon(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+
+        if url != "https://test.example.com/":  # skip during the test, we will test the method separately
```

Comment:
Yeah, not an easy solution.

I have tried to find a solution between existing scrapers, I found one in GoustoJson, which uses `if not settings.TEST_MODE` but it will fail if tests run with `--online` flag.

This solution should be slightly better, but still fails if run with `--online` flag, for some other reason not related to my changes (I guess).

Reply:
Agreed, and good catch about the online flag.  I'll file an issue soon to describe the problems.

Reply:
A possible solution would be to allow setting custom arguments from the test to a scraper's constructor.

E.g. in this case, `json` argument similar to existing `html` can be used

Reply:
Something about the `html` argument is a slight anti-pattern, I think.  I won't be certain about that until we look at solutions for #550 in detail, but my sense is that the scraper code should simply act as if it is requesting content from the web, and the `responses` library will return the appropriate data in each test case (or not, when the test is running `online`).